### PR TITLE
feat(Core/Random): port random system from TrinityCore

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- Port random system from TrinityCore
- Delete inherited `ACE_TSS`

## Issues Addressed:
- Part #3451

## Tests Performed:
-

## How to Test the Changes:
- Test game elements with random

## Target Branch(es):
- [x] Master

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/AI/ScriptedAI/ScriptedCreature.h [hunk 1]
```diff
@@ -100,24 +100,26 @@ class SummonList
...
+        for (auto const& guid : listCopy)
         {
-            Creature* summon = ObjectAccessor::GetCreature(*me, *i);
-            if (summon)
+            Creature* summon = ObjectAccessor::GetCreature(*me, guid);
+            if (summon && summon->IsAIEnabled)
             {
-                if (summon->IsAIEnabled)
-                    summon->AI()->DoAction(info);
+                summon->AI()->DoAction(info);
```
Comment (FrancescoBorzi) [inline src/server/game/AI/ScriptedAI/ScriptedCreature.h:117]: is this change intended? because the behaviour will change here (it will get to the `else` block even when `summon` is defined but `summon->IsAIEnabled` is `false`, unlike before where it only got to the `else` block only when `summon` was not defined

Comment (Winfidonarleyan) [reply src/server/game/AI/ScriptedAI/ScriptedCreature.h:117]: ah yes, nice catch, I forgot about the deletion

**Comment by FrancescoBorzi** [comment]
@azerothcore/testers @sedatyf @mpfans @pklloveyou please check this

**Comment by mpfans** [comment]
How to test?

**Comment by Sedatyf** [comment]
> How to test?

I was just about to ask that same question. What we're looking here?
Is it like https://github.com/azerothcore/azerothcore-wotlk/pull/5333?

**Comment by Winfidonarleyan** [comment]
> > How to test?
> 
> I was just about to ask that same question. What we're looking here?
> Is it like #5333?

Yes, test random system in game (example loot)

**Comment by Sedatyf** [comment]
Hey there,
I wanted to test this PR but I had problems building it. I'm Windows 10 and I created a branch for this PR from master
image

Then I deleted my ``build`` folder, run again CMake, and finally, launched Visual build. At the end, I have those errors:
image

It seems @r-o-b-o-t-o has the same problem as me when he tries to build this PR. But I'm not how CI built this to be honest
Any insight?

**Comment by Winfidonarleyan** [comment]
```sh
git reset --hard 91931504a2a4fac9c0e31d6345b08ac52a86aba9
```

**Comment by Sedatyf** [comment]
Hey @Winfidonarleyan 
I got same errors. I did what I previously posted on the first screen. Then ``git reset`` then run CMake and Visual and I still have errors
image

**Comment by Sedatyf** [comment]
I forgot to mention it before but your fix worked @Winfidonarleyan and I could test your PR.

I played nearly one hour. I did instances in group with another character. So I could roll on items. I tested greed, need, different looting rules (master looter, need before greed, etc.). I looted chests and mobs. Everything looked fine. There wasn't any error in console and server didn't crash

**Comment by Sedatyf** [comment]
If you want me to test more I can check other thing more specific. Just let me know!

**Review by FrancescoBorzi — APPROVED** [review]
